jaeger-core/src/test/java/com/uber/jaeger/baggage/BaggageSetterTest.java

```diff
@@ -0,0 +1,116 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.uber.jaeger.baggage;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+import com.uber.jaeger.LogData;
+import com.uber.jaeger.Span;
+import com.uber.jaeger.SpanContext;
+import com.uber.jaeger.Tracer;
+import com.uber.jaeger.metrics.InMemoryStatsReporter;
+import com.uber.jaeger.metrics.Metrics;
+import com.uber.jaeger.metrics.StatsFactoryImpl;
+import com.uber.jaeger.reporters.InMemoryReporter;
+import com.uber.jaeger.samplers.ConstSampler;
+
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class BaggageSetterTest {
+
+  private InMemoryReporter reporter;
+  private Tracer tracer;
+  private Span span;
+  private InMemoryStatsReporter metricsReporter;
+  private Metrics metrics;
+
+  private static final String KEY = "key";
+
+  @Before
+  public void setUp() throws Exception {
+    metricsReporter = new InMemoryStatsReporter();
+    reporter = new InMemoryReporter();
+    metrics = new Metrics(new StatsFactoryImpl(metricsReporter));
+    tracer =
+        new Tracer.Builder("SamplerTest", reporter, new ConstSampler(true))
+            .withStatsReporter(metricsReporter)
+            .withBaggageRestrictionManager(new DefaultBaggageRestrictionManager(metrics, 15))
+            .build();
+    span = (Span) tracer.buildSpan("some-operation").startManual();
+  }
+
+  @Test
+  public void testInvalidBaggage() {
+    BaggageSetter setter = BaggageSetter.of(false, 0, metrics);
+    final String value = "value";
+    SpanContext ctx = setter.setBaggage(span, KEY, value);
+
+    assertBaggageLogs(span, KEY, value, false, true);
+    assertNull(ctx.getBaggageItem(KEY));
+
+    assertEquals(
+        1L, metricsReporter.counters.get("jaeger.baggage-update.result=err").longValue());
+  }
+
+  @Test
+  public void testTruncatedBaggage() {
+    BaggageSetter setter = BaggageSetter.of(true, 5, metrics);
+    final String value = "0123456789";
+    final String expected = "01234";
+    SpanContext ctx = setter.setBaggage(span, KEY, value);
+
+    assertBaggageLogs(span, KEY, expected, true, false);
+    assertEquals(expected, ctx.getBaggageItem(KEY));
+
+    assertEquals(
+        1L, metricsReporter.counters.get("jaeger.baggage-truncate").longValue());
+    assertEquals(
+        1L, metricsReporter.counters.get("jaeger.baggage-update.result=ok").longValue());
+  }
+
+  private void assertBaggageLogs(
```

Comment:
It looks like testing concerns aren't separated between SpanTest and BaggageSetter.

Ideally, we would only do the following in SpanTest
Assert that `span.setBaggageItem` invokes `BaggageSetter` with the right arguments. 
Assert that `span.getBaggeItem` can retrieve baggage that has been set

All of the detailed test cases that only involve setting baggage should be under the purview of `BaggageSetterTest`. This results in a clear separation of concerns and code deduplication.